src/sentry/static/sentry/app/views/stream/stream.jsx

```diff
@@ -701,6 +701,11 @@ const Stream = createReactClass({
   renderStreamBody() {
     let body;
     let project = this.getProject();
+
+    if (project.firstEvent) {
+      ConfigStore.set('sentFirstEvent', project.firstEvent);
+    }
```

Comment:
I'm a little late to the game here, but we should try and fix this at some point.

As [noted here in the react docs](https://reactjs.org/docs/react-component.html#reference):

> The render() function should be pure, meaning that it does not modify component state, it returns the same result each time it’s invoked, and it does not directly interact with the browser.

Updating the `ConfigStore` is actually a side effect, making this an unpure function, and potentially causing bad side effects.

It should likely live in `componentDidUpdate`

Reply:
Also I'm not quite sure when project will be updated with the first event.